pkg/flowkit/services/transactions.go

```diff
@@ -223,60 +288,52 @@ func (t *Transactions) SendSigned(
 
 // Send a transaction code using the signer account and arguments for the specified network.
 func (t *Transactions) Send(
-	signer *flowkit.Account,
-	code []byte,
-	codeFilename string,
+	accounts *transactionAccounts,
+	script *Script,
 	gasLimit uint64,
-	args []cadence.Value,
 	network string,
 ) (*flow.Transaction, *flow.TransactionResult, error) {
 	if t.state == nil {
 		return nil, nil, fmt.Errorf("missing configuration, initialize it: flow state init")
 	}
 
-	signerKeyIndex := signer.Key().Index()
-
 	tx, err := t.Build(
-		signer.Address(),
-		[]flow.Address{signer.Address()},
-		signer.Address(),
-		signerKeyIndex,
-		code,
-		codeFilename,
+		accounts.toAddresses(),
+		accounts.proposer.Key().Index(),
+		script,
 		gasLimit,
-		args,
 		network,
-		true,
 	)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	err = tx.SetSigner(signer)
-	if err != nil {
-		return nil, nil, err
-	}
+	for _, signer := range accounts.getSigners() {
+		err = tx.SetSigner(signer)
+		if err != nil {
+			return nil, nil, err
+		}
 
-	signed, err := tx.Sign()
-	if err != nil {
-		return nil, nil, err
+		tx, err = tx.Sign()
+		if err != nil {
+			return nil, nil, err
+		}
 	}
 
-	t.logger.Info(fmt.Sprintf("Transaction ID: %s", signed.FlowTransaction().ID()))
+	t.logger.Info(fmt.Sprintf("Transaction ID: %s", tx.FlowTransaction().ID()))
 	t.logger.StartProgress("Sending transaction...")
-	defer t.logger.StopProgress()
 
-	sentTx, err := t.gateway.SendSignedTransaction(signed)
+	sentTx, err := t.gateway.SendSignedTransaction(tx)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	t.logger.StopProgress()
 	t.logger.StartProgress("Waiting for transaction to be sealed...")
+	defer t.logger.StopProgress()
```

Comment:
how about:
```
func withLoggerProgress(logger, msg string ,f func() ) {
   logger.StartProgress(msg)
   defer logger.StopProgress()
   f()
}
```

Reply:
named return values are an option, but kills the readability. Maybe something like scoped progress with finalizer can be useful.

Reply:
> scoped progress with finalizer can be useful.

Can you elaborate? meaning returning a func that stops the progress?

Reply:
hmm nvm, I thought a bit like cpp.